File: apps/backend/app/schemas/wallet.py
# backend/app/schemas/wallet.py
from uuid import UUID
from pydantic import BaseModel, ConfigDict, Field
from decimal import Decimal
from datetime import datetime
from typing import Optional, List
from enum import Enum

class WalletRead(BaseModel):
    company_id: UUID
    balance: Decimal
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

class UserCashbackWalletRead(BaseModel):
    id: UUID
    user_id: UUID
    company_id: UUID
    balance: float
    created_at: datetime
    updated_at: datetime
    company_name: Optional[str] = None
    company_logo_url: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)

class WalletSummary(BaseModel):
    total_balance: float = 0.0
    wallet_count: int = 0

    model_config = ConfigDict()

class UserWalletRead(BaseModel):
    id: UUID
    user_id: UUID
    company_id: UUID
    balance: Decimal

    model_config = ConfigDict(from_attributes=True)

class WalletWithdraw(BaseModel):
    amount: Decimal = Field(..., gt=0, description="Valor a debitar da carteira do usuário")

    model_config = ConfigDict()


class WalletTransactionRead(BaseModel):
    id: UUID
    company_id: UUID
    user_id: UUID
    amount: Decimal
    created_at: datetime 

    model_config = ConfigDict(from_attributes=True)





class WalletOperation(BaseModel):
    amount: Decimal = Field(..., gt=0)
    description: Optional[str] = None

    model_config = ConfigDict()


class CreditTxType(str, Enum):
    CREDIT = "credit"
    DEBIT  = "debit"

class WalletTransactionRead(BaseModel):
    id: UUID
    type: CreditTxType
    amount: Decimal
    description: Optional[str]
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class PaginatedWalletTransactions(BaseModel):
    total: int
    skip: int
    limit: int
    items: List[WalletTransactionRead]

    model_config = ConfigDict(from_attributes=True)

File: apps/portal-client/src/types/wallet.ts
// src/types/wallet.ts

export interface UserCashbackWalletRead {
  id: string;
  user_id: string;
  company_id: string;
  balance: number;
  created_at: string;    // ISO date string
  updated_at: string;    // ISO date string
  company_name?: string;
  company_logo_url?: string;
}

export interface WalletSummary {
  total_balance: number;
  wallet_count: number;
}

export interface WalletTransactionRead {
  id: string;
  company_id: string;
  user_id: string;
  amount: number;       // ou string, dependendo de como vem do back
  created_at: string;   // ISO date string
}

export interface WalletRead {
  company_id: string;   // UUID
  balance: number;      // Decimal convertido para number
  created_at: string;   // ISO datetime
  updated_at: string;   // ISO datetime
}


export interface WalletOperation {
  amount: number;          // valor em reais
  description?: string;    // texto opcional
}

/**
 * Transação de crédito/débito na carteira (admin)
 */
export interface AdminWalletTransaction {
  id: string;
  type: 'credit' | 'debit';
  amount: number;
  description?: string;
  created_at: string;      // ISO date string
}

/**
 * Extrato paginado de transações da carteira (admin)
 */
export interface PaginatedWalletTransactions {
  total: number;
  skip: number;
  limit: number;
  items: AdminWalletTransaction[];
}